File: 0974. Subarray Sums Divisible by K/Solution0974.cpp
class Solution0974 {
public:
    int subarraysDivByK(vector<int>& nums, int k) {
        vector<int> rem(k, 0);
        rem[0] = 1;
        int curSum = 0, count = 0, curRem = 0;
        for (int i = 0; i < nums.size(); i++) {
            curSum += nums[i];
            if (curSum % k >= 0) {
                curRem = curSum % k;
            } else {
                curRem = curSum % k + k;
            }
            count += rem[curRem];
            rem[curRem]++;
        }
        return count;
    }
};

File: 0974. Subarray Sums Divisible by K/Solution0974.go
func subarraysDivByK(nums []int, k int) int {
	curSum, curRem, count := 0, 0, 0
	rem := map[int]int{0: 1}         // !!! init zero array as map
	for i := 0; i < len(nums); i++ { // for _,num:=range nums{
		curSum += nums[i]
		curRem = curSum % k
		if curRem < 0 {
			curRem += k
		}
		count += rem[curRem]
		rem[curRem] += 1
	}
	return count
}